tests/zerotier/test_parser.py

```diff
@@ -0,0 +1,327 @@
+                "id": "9536600adf654321",
+                "nwid": "9536600adf654321",
+                "objtype": "network",
+                "revision": 1,
+                "creationTime": 1632012345,
+                "name": "zerotier-openwisp-network",
+                "private": True,
+                "enableBroadcast": True,
+                "v4AssignMode": {"zt": True},
+                "v6AssignMode": {"6plane": False, "rfc4193": True, "zt": True},
+                "mtu": 2700,
+                "multicastLimit": 16,
+                "routes": [{"target": "10.0.0.0/24", "via": "10.0.0.1"}],
+                "ipAssignmentPools": [
+                    {"ipRangeStart": "10.0.0.10", "ipRangeEnd": "10.0.0.100"}
+                ],
+                "dns": {"domain": "zerotier.openwisp.io", "servers": ["10.147.20.3"]},
+                "rules": [
+                    {
+                        "etherType": 2048,
+                        "not": True,
+                        "or": False,
+                        "type": "MATCH_ETHERTYPE",
+                    },
+                    {"type": "ACTION_DROP"},
+                ],
+                "capabilities": [
+                    {
+                        "default": True,
+                        "id": 1,
+                        "rules": [
+                            {
+                                "etherType": 2048,
+                                "not": True,
+                                "or": False,
+                                "type": "MATCH_ETHERTYPE",
+                            }
+                        ],
+                    }
+                ],
+                "tags": [{"default": 1, "id": 1}],
+                "remoteTraceTarget": "7f5d90eb87",
+                "remoteTraceLevel": 1,
+            }
+        ]
+    }
+
+    # Multiple test config
+    _TEST_MULTIPLE_CONFIG = deepcopy(_TEST_CONFIG)
+    _TEST_MULTIPLE_CONFIG['zerotier'].append(
+        {
+            "id": "9536600adf654322",
+            "nwid": "9536600adf654322",
+            "objtype": "network",
+            "revision": 1,
+            "creationTime": 1632012345,
+            "name": "zerotier-openwisp-network-2",
+            "private": True,
+            "enableBroadcast": True,
+            "v4AssignMode": {"zt": True},
+            "v6AssignMode": {"6plane": False, "rfc4193": True, "zt": True},
+            "mtu": 2700,
+            "multicastLimit": 16,
+            "routes": [{"target": "10.0.0.0/24", "via": "10.0.0.1"}],
+            "ipAssignmentPools": [
+                {"ipRangeStart": "10.0.0.10", "ipRangeEnd": "10.0.0.100"}
+            ],
+            "dns": {"domain": "zerotier.openwisp.io", "servers": ["10.147.20.3"]},
+            "tags": [{"default": 1, "id": 1}],
+            "remoteTraceTarget": "7f5d90eb87",
+            "remoteTraceLevel": 1,
+        }
+    )
+
+    def test_parse_text(self):
+        native = """// zerotier controller config: 9536600adf654321.json
```

Comment:
I am a bit perplexed here, the point of the parse methods is to be able to read the native configuration format of the tool we are integrating. Can zerotier allow to download its configuration in JSON format?

Reply:
It is possible to retrieve network configuration in JSON format using REST API endpoints. But, that would not include the comment there.

Reply:
@nemesisdesign, we can obtain a [dump of ZeroTier settings](https://pastebin.com/mA42045S) using the `zerotier-cli dump` command. However, unfortunately, this dump does not include the JSON network configuration from the **network controller**. Therefore, our only option is to retrieve the configuration either through the REST API or by directly accessing the copy stored in the `/var/lib/zerotier-one/controller.d/network` directory.

Reply:
Yes, the parser would [ignore any comment in the configuration](https://github.com/openwisp/netjsonconfig/pull/272/files#diff-4e782cb854b8f20ae055ed735e67603137d6014ed8a1f73f8ba00b7ca107a1baR33-R38).

Reply:
Are you suggesting that I should [remove the comment](https://github.com/openwisp/netjsonconfig/pull/272/files#diff-1ad7f38c2f702da652381ecfb5a3ddae744d893f0f84cf61ca6a7ba2e7c71bc8R2) from the ZeroTier template? @pandafy @nemesisdesign

Reply:
Yes. The test should emulate real world scenarios. You can remove the comment similar tests.

Reply:
@Aryamanz29 can you make sure the parser can parse both the version with comments (ignore any comment) and the version without comments?

Something like this: https://stackoverflow.com/a/69022887 (I haven't tested it).

Reply:
@nemesisdesign Added in [`23f338f` (#272)](https://github.com/openwisp/netjsonconfig/pull/272/commits/23f338fac3330d33bb52ac485fca3c3bb89c2ad7)